packages/x-tree-view/src/internals/plugins/useTreeViewLabel/useTreeViewLabel.types.ts

```diff
@@ -0,0 +1,74 @@
+import { TreeViewPluginSignature } from '../../models';
+import { TreeViewItemId } from '../../../models';
+import { UseTreeViewItemsSignature } from '../useTreeViewItems';
+
+export interface UseTreeViewLabelPublicAPI {
+  /**
+   * Used to update the label of an item.
+   * @param {TreeViewItemId} itemId The id of the item that is being edited.
+   * @param {string} newLabel The new label of the item.
+   */
+  updateItemLabel: (itemId: TreeViewItemId, newLabel: string) => void;
+}
+
+export interface UseTreeViewLabelInstance extends UseTreeViewLabelPublicAPI {
+  /**
+   * Updates which item is currently being edited.
+   * @param {TreeViewItemId} itemId The id of the item that is currently being edited.
+   * @returns {void}.
+   */
+  setEditedItemId: (itemId: TreeViewItemId | null) => void;
+  /**
+   * Checks if an item is being edited or not.
+   * @param {TreeViewItemId} itemId The id of the item to check.
+   * @returns {boolean}.
+   */
+  isItemBeingEdited: (itemId: TreeViewItemId) => boolean;
+  /**
+   * Checks if an item is being edited or not.
+   * Purely internal use, used to avoid unnecessarily calling `updateItemLabel` twice.
+   * @param {TreeViewItemId} itemId The id of the item to check.
+   * @returns {boolean}.
+   */
+  isItemBeingEditedRef: (itemId: TreeViewItemId) => boolean;
+  /**
+   * Determines if a given item is editable.
+   * @param {TreeViewItemId} itemId The id of the item to check.
+   * @returns {boolean} `true` if the item is editable.
+   */
+  isItemEditable: (itemId: TreeViewItemId) => boolean;
+  /**
+   * Set to `true` if the tree view is editable.
+   */
+  isTreeViewEditable: boolean;
+}
+
+export interface UseTreeViewLabelParameters<R extends {}> {
+  /**
+   * Callback fired when the label of an item changes.
+   * @param {TreeViewItemId} itemId The id of the item that was edited.
+   * @param {string} newLabel The new label of the items.
+   */
+  onItemLabelChange?: (itemId: TreeViewItemId, newLabel: string) => void;
+  /**
+   * Determines if a given item is editable or not.
+   * The items are not editable by default.
```

Comment:
It's more common to put the "by default" at the beginning 

```suggestion
   * By default, items are not editable.
```

Or could use a decorator

```suggestion
   * @default (item) => false
```

Reply:
For some reason in this case I feel like it makes more sense without the decorator - but I can't explain it tbh so I'm very flexible 😅 
I'm okay with both of your suggestions 👌

Reply:
I would not use a decorator here since it's the default value of the item, it's not the default value of the method (which is always defined).

I agree that putting the `By default` at the beginning sounds like the right choice